groups/nts/ntsu/ntsu_socketutil.cpp

```diff
@@ -416,6 +421,138 @@ void OutoingDataStats::update(const struct iovec* buffersSendable,
 
 #endif
 
+#if NTSU_SOCKETUTIL_DEBUG_RECVMSG
+
+class IncomingDataStats
+{
+    // This class tracks the statistics for 'sendmsg'.
+
+    bsls::AtomicUint64 d_numSyscalls;
+    bsls::AtomicUint64 d_totalBuffersReceivable;
+    bsls::AtomicUint64 d_totalBytesReceivable;
+    bsls::AtomicUint64 d_totalBytesReceived;
+    bsls::AtomicUint64 d_numWouldBlocks;
+
+  public:
+    IncomingDataStats();
+
+    ~IncomingDataStats();
+
+    void update(const struct iovec* buffersReceivable,
+                bsl::size_t         numBuffersReceivable,
+                ssize_t             numBytesSent,
+                int                 errorNumber);
+};
+
+IncomingDataStats s_incomingDataStats;
+
+IncomingDataStats::IncomingDataStats()
+: d_numSyscalls(0)
+, d_totalBuffersReceivable(0)
+, d_totalBytesReceivable(0)
+, d_totalBytesReceived(0)
+, d_numWouldBlocks(0)
+{
+}
+
+IncomingDataStats::~IncomingDataStats()
+{
+    if (d_numSyscalls == 0) {
+        return;
+    }
+
+    double avgBuffersReceivablePerSyscall = 0;
+    if (d_numSyscalls > 0) {
```

Comment:
This check seems to be useless due to the condition above. This comment is applicable below and for `~OutgoingDataStats()` code